include/ruby/internal/core/rmatch.h

```diff
@@ -101,17 +101,21 @@ struct RMatch {
      */
     VALUE str;
 
-    /**
-     * The result of this match.
-     */
-    struct rmatch *rmatch;
-
     /**
      * The expression of this match.
      */
     VALUE regexp;  /* RRegexp */
 };
 
+typedef struct rb_matchext_struct {
+    /**
+     * The result of this match.
+     */
+    struct rmatch rmatch;
+} rb_matchext_t;
```

Comment:
Instead of wrapping the `struct rmatch` inside of `struct rb_matchext_struct`, can we just allocate the `rmatch` after the `RMatch`? Saves one level of wrapping.

Reply:
Can we then rename `struct rmatch` (which, IMO, is a terrible name because it's confusing with `struct RMatch`) to `struct rb_matchext_struct`? We can then just add fields to `struct rb_matchext_struct` in the future if we want to add more fields.

Reply:
> Can we then rename `struct rmatch` (which, IMO, is a terrible name because it's confusing with `struct RMatch`) to `struct rb_matchext_struct`? We can then just add fields to `struct rb_matchext_struct` in the future if we want to add more fields.

Good idea.  I also feel it confusing to have both `struct rmatch` and `struct RMatch`.